client/js/widgets/button.js

```diff
@@ -58,6 +64,7 @@ class Button extends Widget {
       const a = { ...original };
       var problems = [];
 
+      if (this.p('debug')) console.log(`${this.id}: ${JSON.stringify(original)}`);
```

Comment:
Seems to be unrelated to the PR. It's fine, I just wonder what you used it for.

Reply:
I found it helpful when debugging my code to have a copy of the original before doing the applyVariables.